src/Factory.sol

```diff
@@ -86,9 +99,13 @@ contract QuestFactory is Ownable2Step, AccessControl {
         string memory _symbol,
         address[] memory _contributors,
         string memory _tokenURI,
-        string memory _contractURI
+        string memory _contractURI,
+        string memory _nonce
     ) external onlyOwnerOrAdmin returns (Quest) {
-        Quest _quest = new Quest(
+        bytes32 salt = keccak256(abi.encodePacked(msg.sender, _nonce));
+        Quest _quest = Quest( Clones.cloneDeterministic(questImplementation, salt) );
```

Comment:
If you don't have a need for knowing the contract addresses ahead of time, you could just use the simpler `clone` instead of `cloneDeterministic`.

It would remove the need for the additional salt, making quest creation a little more straight-forward.

```diff
-        bytes32 salt = keccak256(abi.encodePacked(msg.sender, _nonce));
-        Quest _quest = Quest( Clones.cloneDeterministic(questImplementation, salt)
+        Quest _quest = Quest( Clones.clone(questImplementation)
```